File: CS/EmployeeAsUserExample.Module/DatabaseUpdate/Updater.cs
using System;

using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Updating;
using DevExpress.Xpo;
using DevExpress.Data.Filtering;
using DevExpress.Persistent.BaseImpl;
using DevExpress.ExpressApp.Security;
using EmployeeAsUserExample.Module.BusinessObjects;

namespace EmployeeAsUserExample.Module.DatabaseUpdate {
    public class Updater : ModuleUpdater {
        public Updater(IObjectSpace objectSpace, Version currentDBVersion) : base(objectSpace, currentDBVersion) { }
        public override void UpdateDatabaseAfterUpdateSchema() {
            base.UpdateDatabaseAfterUpdateSchema();
            EmployeeRole adminEmployeeRole = ObjectSpace.FindObject<EmployeeRole>(
                new BinaryOperator("Name", SecurityStrategy.AdministratorRoleName));
            if (adminEmployeeRole == null) {
                adminEmployeeRole = ObjectSpace.CreateObject<EmployeeRole>();
                adminEmployeeRole.Name = SecurityStrategy.AdministratorRoleName;
                adminEmployeeRole.IsAdministrative = true;
            }
            Employee adminEmployee = ObjectSpace.FindObject<Employee>(
                new BinaryOperator("UserName", "Administrator"));
            if (adminEmployee == null) {
                adminEmployee = ObjectSpace.CreateObject<Employee>();
                adminEmployee.UserName = "Administrator";
                adminEmployee.FirstName = "Andrew";
                adminEmployee.LastName = "Fuller";
                adminEmployee.SetPassword("");
                adminEmployee.EmployeeRoles.Add(adminEmployeeRole);
            }
            EmployeeTask task1 = ObjectSpace.FindObject<EmployeeTask>(
                new BinaryOperator("Subject", "Sample task that belongs to Andrew"));
            if (task1 == null) {
                task1 = ObjectSpace.CreateObject<EmployeeTask>();
                task1.Subject = "Sample task that belongs to Andrew";
                task1.Owner = adminEmployee;
            }
            EmployeeTask task2 = ObjectSpace.FindObject<EmployeeTask>(
                new BinaryOperator("Subject", "Sample task that belongs to nobody"));
            if (task2 == null) {
                task2 = ObjectSpace.CreateObject<EmployeeTask>();
                task2.Subject = "Sample task that belongs to nobody"; 
            }    
            ObjectSpace.CommitChanges();
        }
    }
}

File: VB/EmployeeAsUserExample.Module/DatabaseUpdate/Updater.vb
Imports System

Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Updating
Imports DevExpress.Xpo
Imports DevExpress.Data.Filtering
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.ExpressApp.Security
Imports EmployeeAsUserExample.Module.BusinessObjects

Namespace EmployeeAsUserExample.Module.DatabaseUpdate
    Public Class Updater
        Inherits ModuleUpdater

        Public Sub New(ByVal objectSpace As IObjectSpace, ByVal currentDBVersion As Version)
            MyBase.New(objectSpace, currentDBVersion)
        End Sub
        Public Overrides Sub UpdateDatabaseAfterUpdateSchema()
            MyBase.UpdateDatabaseAfterUpdateSchema()
            Dim adminEmployeeRole As EmployeeRole = ObjectSpace.FindObject(Of EmployeeRole)(New BinaryOperator("Name", SecurityStrategy.AdministratorRoleName))
            If adminEmployeeRole Is Nothing Then
                adminEmployeeRole = ObjectSpace.CreateObject(Of EmployeeRole)()
                adminEmployeeRole.Name = SecurityStrategy.AdministratorRoleName
                adminEmployeeRole.IsAdministrative = True
            End If
            Dim adminEmployee As Employee = ObjectSpace.FindObject(Of Employee)(New BinaryOperator("UserName", "Administrator"))
            If adminEmployee Is Nothing Then
                adminEmployee = ObjectSpace.CreateObject(Of Employee)()
                adminEmployee.UserName = "Administrator"
                adminEmployee.FirstName = "Andrew"
                adminEmployee.LastName = "Fuller"
                adminEmployee.SetPassword("")
                adminEmployee.EmployeeRoles.Add(adminEmployeeRole)
            End If
            Dim task1 As EmployeeTask = ObjectSpace.FindObject(Of EmployeeTask)(New BinaryOperator("Subject", "Sample task that belongs to Andrew"))
            If task1 Is Nothing Then
                task1 = ObjectSpace.CreateObject(Of EmployeeTask)()
                task1.Subject = "Sample task that belongs to Andrew"
                task1.Owner = adminEmployee
            End If
            Dim task2 As EmployeeTask = ObjectSpace.FindObject(Of EmployeeTask)(New BinaryOperator("Subject", "Sample task that belongs to nobody"))
            If task2 Is Nothing Then
                task2 = ObjectSpace.CreateObject(Of EmployeeTask)()
                task2.Subject = "Sample task that belongs to nobody"
            End If
            ObjectSpace.CommitChanges()
        End Sub
    End Class
End Namespace
